Scenes/EncountersMenuScene.gd

```diff
@@ -165,48 +167,46 @@ func _run():
 		for chance in [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.2, 1.5, 2.0, 3.0]:
 			addButton(str(Util.roundF(chance*100.0))+"%", "Pick this chance", "setspecieschance", [species, chance])
 
-	if(state == "forgetmenu"):
+	if(state == "npclistmenu"):
+		GM.ui.showNpcList()
 		var encounterPools = GM.main.getDynamicCharactersPools()
 
-		saynn("Select which occupation the character that you wanna forget has.")
+		saynn("Select which occupation the character that you want to look for has.")
 
-		addButton("Back", "Go back a level", "")
+		addButton("Back", "Go back a level", "") 
 
 		for encounterPoolID in encounterPools:
-			addButton(str(encounterPoolID), "Pick this occupation", "forgetmenupool", [encounterPoolID])
-		
-		
-	if(state == "forgetmenupool"):
-		saynn("Pick who do you wanna forget. They will never show up again. This action can not be undone.")
-		
-		saynn("Keep in mind that if this character is pregnant, their pregnancy will be forgotten too. But any kids you had together will stay.")
+			addButton(str(encounterPoolID), "Pick this occupation", "occupationmenupool", [encounterPoolID])
 		
-		addButton("Back", "Go back a level", "forgetmenu")
 		
-		var characterIDS = GM.main.getDynamicCharacterIDsFromPool(pickedPoolToForget)
+	if(state == "occupationmenupool"):
+		GM.ui.npcListScreen.clearRows()
+
+		var characterIDS = GM.main.getDynamicCharacterIDsFromPool(pickedPoolToShow)
 		for characterID in characterIDS:
-			var dynamicCharacter = GlobalRegistry.getCharacter(characterID)
+			var dynamicCharacter: BaseCharacter  = GlobalRegistry.getCharacter(characterID)
 			if(dynamicCharacter == null):
 				continue
-				
-			var desc = dynamicCharacter.getSmallDescription()
-			if(desc == null):
-				desc = ""
-			
-			var kidsAmount = GM.CS.getChildrenAmountOf(characterID)
-			if(kidsAmount > 0):
-				if(desc != ""):
-					desc += "\n"
-				desc += dynamicCharacter.getName()+" has "+str(kidsAmount)+" "+Util.multipleOrSingularEnding(kidsAmount, "kid")+"."
-			
+
+#			var desc = dynamicCharacter.getSmallDescription()
+#			if(desc == null):
+#				desc = ""
+
+#			var kidsAmount = GM.CS.getChildrenAmountOf(characterID)
+#			if(kidsAmount > 0):
+#				if(desc != ""):
+#					desc += "\n"
+
+			var NPCname = dynamicCharacter.getName()
+			var gender = dynamicCharacter.npcGeneratedGender.capitalize() #accessing param like this is kinda wrong but getGender() returns number and I don't want that
+			var subbyStat = dynamicCharacter.getPersonality().getStat(PersonalityStat.Subby)
+			var personality = PersonalityStat.getVisibleDesc("Subby", subbyStat)
 			var sharedKidsAmount = GM.CS.getSharedChildrenAmount("pc", characterID)
-			if(sharedKidsAmount > 0):
-				if(sharedKidsAmount == kidsAmount):
-					desc += " All of them are from you."
-				else:
-					desc += " "+str(sharedKidsAmount)+" of them are from you."
+
+			GM.ui.npcListScreen.addRow(NPCname, gender, personality, characterID, sharedKidsAmount)
```

Comment:
this would become 
`npclist.addRow(NPCname, gender, personality, characterID, sharedKidsAmount)`